static/compiler-picker.js

```diff
@@ -0,0 +1,188 @@
+    this.lastLangId = langId;
+    this.lastCompilerId = compilerId;
+
+    this.tomSelect = new TomSelect(this.domNode, {
+        sortField: this.compilerService.getSelectizerOrder(),
+        valueField: 'id',
+        labelField: 'name',
+        searchField: ['name'],
+        optgroupField: '$groups',
+        optgroups: this.getGroups(langId),
+        lockOptgroupOrder: true,
+        options: this.getOptions(langId, compilerId),
+        items: compilerId ? [compilerId] : [],
+        dropdownParent: 'body',
+        closeAfterSelect: true,
+        plugins: ['dropdown_input'],
+        maxOptions: 1000,
+        onChange: _.bind(function (val) {
+            if (val) {
+                ga.proxy('send', {
+                    hitType: 'event',
+                    eventCategory: 'SelectCompiler',
+                    eventAction: val,
+                });
+                this.onCompilerChange(val);
+            }
+        }, this),
+        duplicates: true,
+        render: {
+            option: function (data, escape) {
+                var extraClasses = data.$groups.indexOf(favoriteGroupName) !== -1 ? ' fav' : '';
+                return '<div class="d-flex' + extraClasses + '"><div>' + escape(data.name) + '</div>' +
+                    '<div title="Click to mark or unmark as a favorite" class="ml-auto toggle-fav">' +
+                    '<i class="fas fa-star"></i>' +
+                    '</div>' +
+                    '</div>';
+            },
+        },
+    });
+
+    $(this.tomSelect.dropdown_content).on('click', '.toggle-fav', _.bind(function (evt) {
+        evt.preventDefault();
+        evt.stopPropagation();
+
+        var optionElement = evt.currentTarget.closest('.option');
+        var clickedGroup = optionElement.parentElement.dataset.group;
+        var value = optionElement.dataset.value;
+        var data = this.tomSelect.options[value];
+        var isAddingNewFavorite = data.$groups.indexOf(favoriteGroupName) === -1;
+        var elemTop = optionElement.offsetTop;
+
+        if (isAddingNewFavorite) {
+            data.$groups.push(favoriteGroupName);
+            this.addToFavorites(data.id);
+        } else {
+            data.$groups.splice(data.group.indexOf(favoriteGroupName), 1);
+            this.removeFromFavorites(data.id);
+        }
+
+        this.tomSelect.updateOption(value, data);
+        this.tomSelect.refreshOptions(false);
+
+        if (clickedGroup !== favoriteGroupName) {
+            // If the user clicked on an option that wasn't in the top "Favorite" group, then we just added
+            // or removed a bunch of controls way up in the list. Find the new element top and adjust the scroll
+            // so the element that was just clicked is back under the mouse.
+            optionElement = this.tomSelect.getOption(value);
+            this.tomSelect.dropdown_content.scrollTop += (optionElement.offsetTop - elemTop);
```

Comment:
> I'm Very annoyed by these movements when selecting and unselecting favorites.

I think the issue is because of the ```scroll-behavior:smooth``` css setting on .ts-dropdown-content

```js
this.tomSelect.dropdown_content.style.scrollBehavior = 'auto';
optionElement = this.tomSelect.getOption(value);
this.tomSelect.dropdown_content.scrollTop += (optionElement.offsetTop - elemTop);
this.tomSelect.dropdown_content.style.scrollBehavior = 'smooth';
```